server/README.md

```diff
@@ -0,0 +1,370 @@
+>   for a short period of time to get a previous value when querying the block.
+>   However this is purely theoretical since block ID is random (hence there should be
+>   no concurrency between clients) and a client considers a block as immutable.
+
+### Invitations
+
+During the process of invitation, one of the client (the claimer) doesn't have a device
+yet (as it is the whole point of the invitation process !).
+Hence using certificates during this step is not possible (since a certificate is signed by a device).
+
+No certificate involvement means:
+
+- Invitations data are totally separated from other data (given other data always relies
+  on certificates to do part of there validation).
+- Invitation request doesn't contains a client-generated timestamp.
+
+This last point is great from a causality point of view:
+
+- With certificates the client propose a change that occurred a given timestamp, then it's
+  up to the server to check for causality to know if it should accept it.
+- Here the client merely ask the server do to something, then the server check it if it
+  possible *now* then do it *now*. No causality can be broken given we only work in the present.
+
+Regarding concurrency, a lock is needed to protect the creation of invitation against:
+
+- Concurrent invitation creations of the same email creating multiple invitations
+  (see [example in Data concurrency paragraph](#data-concurrency)).
+- Concurrent invitation creation and user creation leading an invitation for an existing user.
+
+### What is locked by who ?
+
+The following array recapitulate which topics each command needs to lock (and in which mode).
+
+API Family    |  command                           | common | realm  | sequester | shamir |
+--------------|------------------------------------|--------|--------|-----------|--------|
+invited       | ping                               |        |        |           |        |
+anonymous     | ping                               |        |        |           |        |
+authenticated | ping                               |        |        |           |        |
+authenticated | events_listen                      |        |        |           |        |
+authenticated | certificate_get                    |        |        |           |        |
+anonymous     | organization_bootstrap             |        |        |           |        |
+authenticated | invite_list                        |        |        |           |        |
+invited       | invite_info                        |        |        |           |        |
+authenticated | invite_new_device                  | read   |        |           |        |
+authenticated | invite_new_user                    | read   |        |           |        |
+authenticated | invite_cancel                      | read   |        |           |        |
+authenticated | invite_1_greeter_wait_peer         |        |        |           |        |
```

Comment:
There will be the new invite command to add here after #7761 is merged

Reply:
I would propose to write a script to update the table as it is highly likely that we will forget about it

Reply:
> I would propose to write a script to update the table as it is highly likely that we will forget about it

There is a script now ! (and it checks more than just that ^^)

> There will be the new invite command

I've added them (because the script forced me :laughing: ), but I have the feeling `invite_greeter_start_greeting_attempt` & `invite_claimer_start_greeting_attempt` also requires some form of manual locking in the style of what is needed when creating an invitation: given the attempt is created by using the user ID as discriminant, but multiple attempts with the same discriminant are allowed in the database (only a single one of them is not removed, but this is not something the database can handle on its own with a unique constraint)

@AureliaDolo  what do you think ?